finesse/hardware/tc4820.py

```diff
@@ -0,0 +1,137 @@
+"""This module provides an interface to TC4820 temperature controllers.
+
+TODO: I've used float throughout this module, but a decimal type may be more appropriate
+and prevent rounding errors.
+"""
+import logging
+from typing import Any
+
+from serial import Serial
+
+
+class TC4820:
+    """An interface for TC4820 temperature controllers."""
+
+    MAX_POWER = 511
+    """The maximum value for the power property."""
+
+    def __init__(
+        self, port: str, baudrate: int = 115200, *serial_args: Any, **serial_kwargs: Any
+    ) -> None:
+        """Create a new TC4820.
+
+        Args:
+            port: Serial port name
+            baudrate: Serial port baudrate
+            serial_args: Extra arguments to Serial constructor
+            serial_kwargs: Extra keyword arguments to Serial constructor
+        """
+        self.serial = Serial(port, baudrate, *serial_args, **serial_kwargs)
+
+    def read_int(self) -> int:
+        """Read a message from the TC4820 and decode the number as a signed integer."""
+        # TODO: Handle errors?
```

Comment:
Most definitely. This application - or more likely the follow up to be flown in a plane - need to be extremely robust, so we will need to implement a way of recovering from errors, when they happen.

Reply:
Hmm, that's true.

There's also another way I've thought of that things could be more robust: currently when the checksum isn't valid an error is raised, but really we should try retransmitting our message in this case (maybe 3 retries or so would be sensible).